### PR TITLE
update base k8s version

### PR DESCRIPTION
**Pull request by pgvishnuram**
k8s version 1.18.17-gke.700 has been removed by GKE team (in release channels)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -115,8 +115,8 @@ resource "google_container_cluster" "primary" {
   */
 
   master_auth {
-    username = "admin"
-    password = random_string.password.result
+    username = ""
+    password = ""
```
Comment (kaxil) [inline main.tf:119]: This looks like an unrelated change, do we need this change?

Comment (pgvishnuram) [reply main.tf:119]: i can remove this completely , basic auth is no more supported

Comment (pgvishnuram) [reply main.tf:119]: basic auth is dropped from 1.19

Comment (kaxil) [reply main.tf:119]: aah ok, I will defer it to @danielhoherd

**Review by kaxil — APPROVED** [review]

**Comment by pgvishnuram** [comment]
@danielhoherd - review this change once

File: variables.tf [hunk 2]
```diff
@@ -9,7 +9,7 @@ variable "dns_managed_zone" {
 }
 
 variable "kube_version_gke" {
-  default     = "1.18.17-gke.700"
+  default     = "1.20.9-gke.1001"
```
Comment (danielhoherd) [inline variables.tf:12]:
```suggestion
  default     = "v1.18.20-gke.901"
```

Comment (pgvishnuram) [reply variables.tf:12]: https://cloud.google.com/kubernetes-engine/docs/release-notes-regular

Comment (danielhoherd) [reply variables.tf:12]: @pgvishnuram [we are not using any release channel](https://github.com/astronomer/google-environments/blob/3af602a0dda66abf4c0f3005c8ad5668379c0b6e/prod/cloud/infra/main.tf#L23). As of [September 7 2021](https://cloud.google.com/kubernetes-engine/docs/release-notes#September_07_2021), version 1.18.20-gke.3001 is still an available control plane version.

Comment (pgvishnuram) [reply variables.tf:12]: daniel can we finalise this

Comment (danielhoherd) [reply variables.tf:12]: I talked to @pgvishnuram about this and we need to bump to 1.20 because that is the oldest version supported by the regular channel. These are not the configs that we use internally. We need to either change the channel to UNSPECIFIED, or bump k8s. It seems to me like bumping k8s is the smaller change that would be expected by people who use this, so that's what we're going with. Good catch on those details @pgvishnuram

File: main.tf (hunk 1 above)
Comment (danielhoherd) [reply main.tf:119]: i think we should leave these fields in here because we can use this module with clusters that are earlier than 1.19.

**Review by danielhoherd — APPROVED** [review]
